sound/hybrid_font.h

```diff
@@ -147,11 +149,14 @@ class HybridFont : public SaberBase {
           swing_player_->set_fade_time(swing_player_->length() - swing_player_->pos());
           swing_player_->FadeAndStop();
           swing_player_.Free();
-          swing_player_ = PlayPolyphonic(&swng);
         }
       }
-      else if (!swing_player_) {
-        swing_player_ = PlayPolyphonic(&swng);
+      if (!swing_player_) {
+        if (fabs(swing_acceleration_) > config_.ProffieOSSlashAccelerationThreshold && slsh.files_found()) {
```

Comment:
why fabs() ?
It doesn't make any sense, because if swing_accelration is negative, it means that the we reached the swing threshold by slowing down?

Reply:
In theory yes.  In practise, it seems like the acceleration is almost always negative.

Reply:
That a problem that we should figure out, not work around.
Maybe the filter factor is too big. Maybe try 0.01 instead of 0.2 or something?

Reply:
I'll give that a go

Reply:
as mentioned, changing the filter_factor doesn't alter the values of swing_acceleration_

I'm not sure why we are seeing negative values but after extensive testing, using the absolute value to test for slashes works perfectly.  If I had to guess, there seems to be an oddity that swing speed slows down to near zero when a swing sound plays, perhaps something is blocking the gyro from reading properly?  In any event, using absolute sidesteps this issue.

Reply:
Unfortunately, it's not enough for the code to work.
It must also make sense, or nobody (including me) will ever be able to change it, understand it or fix bugs in it.

Using fabs() doesn't make sense.
Let's say we plot swing speed over time. Then the swing threshold is a line parallel to the X axis. Once the swing speed goes over that line, we trigger a swing or slash.  At the point in time when we go over the line, the slope of the curve MUST BE POSITIVE, because otherwise the swing speed would be going down, not up.

Sidestepping the issue just isn't ok.

Reply:
Ok, I've reworked the start swing function and I've got slashes triggering predictably now, but there were a bunch of changes to make it work.